### PR TITLE
IA-2900: registry view permission

### PR DESCRIPTION
**Pull request by beygorghor**
With this permission, users would not be able to:

- see other Forms than the Reference forms
- edit form submissions via Enketo
- consult the Org unit details
- download data

Therefore the corresponding buttons should be hidden from the interface

Related JIRA tickets : IA-2900

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [x] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- adding read and write permissions
- adapt backend to filter instances and get only reference instances
- display /hide buttons regarding permissions
- use `DisplayIfUserHasPerm`
- add strict mode to `DisplayIfUserHasPerm`and create a new `userHasAllPermissions`function

## How to test

You need a non superadmin user having only registry_read rights.
Connect to a registry page with this user and check that buttons are not present

## Print screen / video

Screenshot 2024-05-22 at 10 46 06

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx [hunk 1]
```diff
@@ -219,12 +228,9 @@ export const Instances: FunctionComponent<Props> = ({
                             columns={tableColumns}
                             count={data?.count ?? 0}
                             params={params}
-                            onTableParamsChange={p =>
-                                dispatch(
-                                    redirectToReplace(baseUrls.registry, p),
-                                )
-                            }
-                            extraProps={{ loading: isFetchingList }}
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/registry/components/Instances.tsx]: ❤️

File: hat/assets/js/apps/Iaso/domains/registry/components/selectedOrgUnit/OrgUnitTitle.tsx [hunk 2]
```diff
@@ -55,24 +57,30 @@ export const OrgUnitTitle: FunctionComponent<Props> = ({ orgUnit, params }) => {
...
-                    <IconButton
-                        url={`${baseUrls.orgUnitDetails}/orgUnitId/${orgUnit.id}`}
-                        color="secondary"
-                        icon="edit"
-                        tooltipMessage={MESSAGES.editOrgUnit}
-                        iconSize="small"
-                        size="small"
-                    />
+                    <DisplayIfUserHasPerm
+                        permissions={[Permissions.REGISTRY_WRITE]}
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/registry/components/selectedOrgUnit/OrgUnitTitle.tsx:61]: Not blocking but maybe needs to be clarified: we use the registry permission to give access to org unit edition?

So if I have the registry read permission and org unit permission, I cannot edit the org unit via the registry page, but I can via the org unit page (and vice-versa)

File: hat/assets/js/apps/Iaso/domains/users/utils.js
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/users/utils.js:1]: ❤️

**Review by quang-le — APPROVED** [review]
Just one general comment about permissions but nothing blocking.

I didn't review the backend though

File: hat/assets/js/apps/Iaso/domains/registry/components/selectedOrgUnit/OrgUnitTitle.tsx (hunk 2 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/registry/components/selectedOrgUnit/OrgUnitTitle.tsx:61]: Screenshot 2024-05-24 at 11 50 18

**Pull request merged**